plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2159,8 +2162,12 @@ public ConnectorAnalyzeMetadata getStatisticsCollectionMetadata(ConnectorSession
         MetadataEntry metadata = handle.getMetadataEntry();
 
         Optional<Instant> filesModifiedAfterFromProperties = getFilesModifiedAfterProperty(analyzeProperties);
+        boolean forceRecalculate = getForceRecalculateStatistics(analyzeProperties).orElse(false);
 
-        Optional<ExtendedStatistics> statistics = statisticsAccess.readExtendedStatistics(session, handle.getLocation());
+        Optional<ExtendedStatistics> statistics = Optional.empty();
+        if (!forceRecalculate) {
```

Comment:
Shouldn't we drop initially the existing stats in case we do a full refresh?
In the unlikely case that `ANALYZE` statement fails,  the stats for the table will still be present.

Reply:
I don't know, but is it bad to have previous stats there  when ANALYZE fails?